### PR TITLE
GatherND_8 reference implementation

### PR DESCRIPTION
**Pull request by achetver**
### Details:
 - Reference implementation for GatherND_8 operation. Methods `visit_attributes` and `clone_with_new_inputs` moved to GatherNDBase class.

### Tickets:
 - 42721
 - 66651

### CODE REVIEW

File: ngraph/core/include/ngraph/op/gather_nd.hpp [hunk 1]
```diff
@@ -4,13 +4,16 @@
 
 #pragma once
 
-#include "ngraph/op/op.hpp"
+#include "ngraph/op/util/gather_nd_base.hpp"
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/gather_nd.hpp:7]: Why did you change this header?

File: ngraph/core/include/ngraph/op/util/gather_nd_base.hpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+#include "openvino/op/util/gather_nd_base.hpp"
+
+namespace ngraph {
+namespace op {
+namespace util {
+using ov::op::util::GatherNDBase;
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/util/gather_nd_base.hpp:13]: I don't think that we need to introduce headers with aliases to some new utility operations which are not included to official OpenVINO opsets.

File: ngraph/core/src/op/gather_nd.cpp [hunk 3]
```diff
@@ -122,3 +66,28 @@ shared_ptr<Node> op::v5::GatherND::clone_with_new_inputs(const OutputVector& new
...
+    NGRAPH_OP_SCOPE(v8_GatherND_visit_attributes);
+    visitor.on_attribute("batch_dims", m_batch_dims);
+    return true;
+}
+
+shared_ptr<Node> op::v8::GatherND::clone_with_new_inputs(const OutputVector& new_args) const {
+    NGRAPH_OP_SCOPE(v8_GatherND_clone_with_new_inputs);
+    check_new_args_count(this, new_args);
+    return make_shared<op::v8::GatherND>(new_args.at(0), new_args.at(1), m_batch_dims);
+}
```
Comment (ilyachur) [inline ngraph/core/src/op/gather_nd.cpp:93]: Looks like we can move these methods to the base class.

File: ngraph/test/backend/gather_nd.in.cpp [hunk 4]
```diff
@@ -479,3 +480,456 @@ NGRAPH_TEST(${BACKEND_NAME}, gather_nd_batch_dims2_lead_dims) {
     EXPECT_TRUE(
         test::all_close_f((vector<float>{2, 5, 11, 13, 19, 23}), read_vector<float>(result), MIN_FLOAT_TOLERANCE_BITS));
 }
+
+//-----v8-----
+NGRAPH_TEST(${BACKEND_NAME}, gather_nd_v8_single_indices) {
```
Comment (ilyachur) [inline ngraph/test/backend/gather_nd.in.cpp:485]: These tests are frozen. Please move these tests to new infrastructure:
https://github.com/openvinotoolkit/openvino/tree/master/docs/template_plugin/tests/functional/op_reference

File: ngraph/test/runtime/ie/unit_test.manifest [hunk 5]
```diff
@@ -1515,6 +1515,23 @@ gather_v7_axis_0_bool
 gather_v7_3d_indices_axis_1_batch_dims_1_int32
 gather_v7_4d_data_axis_2_batch_dims_1_int32
 
+# not yet implemented on CPU/GPU GatherND_8
+gather_nd_v8_single_indices
```
Comment (ilyachur) [inline ngraph/test/runtime/ie/unit_test.manifest:1519]: Please use SLT instead

Comment (achetver) [reply ngraph/test/runtime/ie/unit_test.manifest]: Now we don't need to mark these tests as failed in plugins.

File: ngraph/core/src/op/gather_nd.cpp (hunk 3 above)
Comment (achetver) [reply ngraph/core/src/op/gather_nd.cpp]: Moved.

File: ngraph/core/include/ngraph/op/gather_nd.hpp (hunk 1 above)
Comment (achetver) [reply ngraph/core/include/ngraph/op/gather_nd.hpp]: Reverted.

File: ngraph/core/include/ngraph/op/util/gather_nd_base.hpp (hunk 2 above)
Comment (achetver) [reply ngraph/core/include/ngraph/op/util/gather_nd_base.hpp]: I used same header as for GatherBase operation. Do I need to remode this intoduction?

**Comment by achetver** [comment]
@pavel-esir, @vgavrilo @popovaan, @ilyachur @GlebKazantaev Please, take a look at this PR.

**Review by vgavrilo — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

File: docs/template_plugin/tests/functional/op_reference/gather_nd.cpp [hunk 6]
```diff
@@ -203,6 +203,188 @@ std::vector<GatherNDParams> generateCombinedParams() {
...
+                    Tensor(IN_ET, {2, 3, 3}, std::vector<T>{
+                            3, 8, 6, 10, 12, 13, 23, 24, 22, 29, 28, 32, 36, 37, 37, 41, 48, 48}),
+                    "gather_8_nd_batch_dims2"),
+            GatherNDParams(
+                    Tensor(IN_ET, {2, 3, 4}, std::vector<T>{1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12,
+                                                            13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24}),
+                    Tensor(element::i32, {2, 3, 1, 1}, std::vector<int32_t>{1, 0, 2, 0, 2, 2}),
+                    2,
+                    Tensor(IN_ET, {2, 3, 1}, std::vector<T>{2, 5, 11, 13, 19, 23}),
+                    "gather_8_nd_batch_dims2_lead_dims"),
```
Comment (rkazants) [inline docs/template_plugin/tests/functional/op_reference/gather_nd.cpp:360]: One more test case it will be very nice to have is `batch_dim=2, data_shape_{2, 3, 4, 5}, indice_shape={2,3, 2, 1} -> output_shape={2,3,2,5}`. In this case you gather slices of shape [5] instead of scalar

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
@achetver, one more test case with slices and batch_dim=2 is nice to add. The other looks good.

File: docs/template_plugin/tests/functional/op_reference/gather_nd.cpp (hunk 6 above)
Comment (achetver) [reply docs/template_plugin/tests/functional/op_reference/gather_nd.cpp:360]: Test was added

**Review by popovaan — APPROVED** [review]

File: ngraph/core/reference/include/ngraph/runtime/reference/gather_nd.hpp [hunk 7]
```diff
@@ -43,7 +43,6 @@ void gather_nd(const T* const params,
                T* const out,
                const Shape& params_shape,
                const Shape& indices_shape,
-               const Shape& out_shape,
```
Comment (apankratovantonp) [inline ngraph/core/reference/include/ngraph/runtime/reference/gather_nd.hpp:46]: Changed reference API Here. Arm Plugin needs to be changed too.

Comment (achetver) [reply ngraph/core/reference/include/ngraph/runtime/reference/gather_nd.hpp]: Change reverted.

**Review by rkazants — APPROVED** [review]
